packages/flutter_tools/lib/src/commands/multi_drive.dart

```diff
@@ -0,0 +1,366 @@
+
+    if (await fs.type(testFile) != FileSystemEntityType.FILE) {
+      printError('Test file not found: $testFile');
+      return 1;
+    }
+
+    if (!argResults['use-existing-app']) {
+      // printStatus('Starting application: ${argResults["target"]}');
+
+      if (getBuildMode() == BuildMode.release) {
+        // This is because we need VM service to be able to drive the app.
+        printError(
+          'Flutter Multi-Device Driver does not support running in release mode.\n'
+          '\n'
+          'Use --profile mode for testing application performance.\n'
+          'Use --debug (default) mode for testing correctness (with assertions).'
+        );
+        return 1;
+      }
+
+      int result = await appsStarter(this);
+      if (result != 0) {
+        printError('Application failed to start. Will not run test. Quitting.');
+        return result;
+      }
+    } else {
+      printStatus('Will connect to already running application instance.');
+    }
+
+    // Check for the existance of a `packages/` directory; pub test does not yet
+    // support running without symlinks.
+    if (!new io.Directory('packages').existsSync()) {
+      Status status = logger.startProgress(
+        'Missing packages directory; running `pub get` (to work around https://github.com/dart-lang/test/issues/327):'
+      );
+      await runAsync(<String>[sdkBinaryName('pub'), 'get', '--no-precompile']);
+      status.stop(showElapsedTime: true);
+    }
+
+    try {
+      return await testRunner(<String>[testFile])
+        .catchError((dynamic error, dynamic stackTrace) {
+          printError('CAUGHT EXCEPTION: $error\n$stackTrace');
+          return 1;
+        });
+    } finally {
+      if (!argResults['keep-app-running'] && !argResults['use-existing-app']) {
+        printStatus('Stopping application instance.');
+        try {
+          await allAppsStopper(this);
+        } catch(error, stackTrace) {
+          // TODO(yjbanov): remove this guard when this bug is fixed: https://github.com/dart-lang/sdk/issues/25862
+          printTrace('Could not stop application: $error\n$stackTrace');
+        }
+      } else {
+        printStatus('Leaving the application running.');
+      }
+    }
+  }
+
+  Future<dynamic> _loadSpecs(String specsPath) async {
+    try {
+      // Read specs file into json format
+      dynamic newSpecs = JSON.decode(await new io.File(specsPath).readAsString());
+      // Get the parent directory of the specs file
+      String rootPath = new io.File(specsPath).parent.absolute.path;
+      // Normalize the 'test-path' in the specs file
+      newSpecs['test-path'] = _normalizePath(rootPath, newSpecs['test-path']);
+      // Normalize the 'app-path' in the specs file
+      newSpecs['devices'].forEach((String deviceID, Map<String, String> value) {
+        value['app-path'] = _normalizePath(rootPath, value['app-path']);
+      });
+      return newSpecs;
+    } on io.FileSystemException catch(e) {
+      print(e);
```

Comment:
Use `printError` instead of `print`, and at least show some meaningful error messages (e.g. `"An error occurred while reading the test specs: $specsPath"`). You should avoid just printing out the exception as is.